lib/mix/lib/mix/tasks/test.ex

```diff
@@ -395,7 +395,7 @@ defmodule Mix.Tasks.Test do
           failures > 0 ->
             System.at_exit(fn _ -> exit({:shutdown, 1}) end)
 
-          excluded == total and option_only_present? ->
+          total == excluded + skipped and option_only_present? ->
```

Comment:
I would say the current behaviour is fine. The reason why we raise here is in case you give a bad name to --only. However, if there are skipped tests, it means the name was found! Maybe we should change the error message below to say "but no tests were found"?

Reply:
Two same test modules with the same options being called by Mix, one has a skipped tag, and one doesn't. and one fails and the other doesn't. Shouldn't it be a bug?

Reply:
I am not sure I understand. Can you please expand?

Reply:
well, They are not "the same" modules, because one has a test tagged with `:skip`.

It set this repo up https://github.com/eksperimental/test_only_bug

tests involved:
- https://github.com/eksperimental/test_only_bug/blob/master/test/test_only_bug_test.exs
- https://github.com/eksperimental/test_only_bug/blob/master/test/test_only_bug_skipped_test.exs

**It does fail**
```console
$ mix test --only ten test/test_only_bug_test.exs 
Including tags: [:ten]
Excluding tags: [:test]

Finished in 0.05 seconds
1 doctest, 2 tests, 0 failures, 3 excluded

Randomized with seed 663958
The --only option was given to "mix test" but no test executed
```

**It does not fail**
```console
$ mix test --only ten test/test_only_bug_skipped_test.exs 
Including tags: [:ten]
Excluding tags: [:test]

*

Finished in 0.07 seconds
1 doctest, 2 tests, 0 failures, 2 excluded, 1 skipped

Randomized with seed 291589
```